skimage/feature/_kadir_brady.py

```diff
@@ -0,0 +1,325 @@
+    base_regions = _detect(image, scales)
+    # pruning based on thresholds
+    regions = _prune(base_regions, saliency_threshold, clustering_threshold, k=15)
+
+    return regions.T
+
+
+def _detect(image, scales):
+    """Find keypoints in the given grayscale image.
+
+    Calculates the Shannon entropy of local image attributes for each region
+    over a range of scales, and weight them based on the difference between
+    the local descriptors around scale-space.
+
+    Parameters
+    ----------
+    image : ndarray
+        Input grayscale image
+    scales : ndarray
+        Scales for key regions
+
+    Returns
+    -------
+    A : (4, N) ndarray of float
+        A 2d array with each column representing 4 values, '(gamma, scale, y, x)'
+        where '(y, x)' are coordinates of the region, 'scale' is the
+        size of salient region, and 'gamma' is the saliency score.
+    """
+
+    entropy, weights, r, c = _saliency_param(image, scales)
+
+    # find local maxima in scale space by selecting second derivatives that are less than zero
+    # second derivative kernel
+    fxx = np.array([1, -2, 1])
+
+    # evaluate second derivative using the kernel
+    d_entropy = np.transpose(ndi.correlate1d(entropy.T, fxx, mode='nearest'))
+    d_entropy[0, :] = 0
+    int_pts = np.nonzero(d_entropy < 0)
+
+    # weighting interest points by scale size times the first derivative of the scale-space function
+    gamma = entropy[int_pts] * weights[int_pts]
+    scale = scales[int_pts[0]]
+    row = r[int_pts[1]]
+    column = c[int_pts[1]]
+
+    return np.array([gamma, scale, row, column])
+
+
+def _prune(candidate_regions, saliency_threshold, v_th, k):
+    """Clusters the detected keypoints.
+
+    It selects highly salient points that have local support
+    i.e. nearby points with similar saliency and scale. Each region is
+    sufficiently distant from all others (in Scale-space regions) to
+    qualify as a separate entity.
+
+    Parameters
+    ----------
+    candidate_regions : (4, n) ndarray of float
+        A 2d array with each column representing gamma,scale,y,x.
+    saliency_threshold : float, optional.
+        Features with saliency score more than this will be considered.
+    v_th : int, optional
+        Variance among the regions should be smaller than this threshold
+        for sufficient clustering.
+    k : int
+        No. of clusters
+
+    Returns
+    -------
+    A : (3, N) ndarray of float
+        A 2d array with each column representing 3 values, '(y, x, scale)'
+        where '(y,x)' are coordinates of the region and 'scale' is the
+        size of corresponding salient region.
+    """
+
+    t_gamma, t_scale, t_r, t_c, D = _salient_regions(candidate_regions, saliency_threshold)
+    gamma, scale, row, column = (np.array([]) for i in range(4))
+    
+    n = max(t_gamma.shape)
+    n_reg = 0
+    # clusters matrix
+    cluster = np.zeros((3, k + 1))
+
+    # pruning process
+    for index in range(n):
```

Comment:
Can you explain what you're doing here? It seems to me that you should be able to simplify the logic quite a bit. For example, the code block for the first region (under the `else`) is the same as the code block for a sufficiently-far region. I'm sure you'd be able to condense both code paths into a single clause.